src/modules/mim/firewall.json

```diff
@@ -1,41 +1,305 @@
+                      "name": "delete",
+                      "enumValue": 2
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "command",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "accept",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "reject",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "drop",
+                      "enumValue": 3
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "type",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "inbound",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "outbound",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "forward",
+                      "enumValue": 3
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "protocol",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "all",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "tcp",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "udp",
+                      "enumValue": 3
+                    },
+                    {
+                      "name": "icmp",
+                      "enumValue": 4
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "source",
+                "schema": "string"
+              },
+              {
+                "name": "sourcePort",
+                "schema": "integer"
+              },
+              {
+                "name": "destination",
+                "schema": "string"
+              },
+              {
+                "name": "destinationPort",
+                "schema": "integer"
+              },
```

Comment:
source and destination can be combined in a single foo and port, direction being indicated by the other settings.

Reply:
Keep them separate and add "Address" to source and destination